tests/test_index.py

```diff
@@ -4,15 +4,85 @@
 from factories import DifferentPageFactory, ExamplePageFactory
 from faker import Faker
 from testapp.models import DifferentPage, ExamplePage
+from wagtail_vector_index.ai_utils.backends.base import BaseEmbeddingBackend
 from wagtail_vector_index.storage import (
     registry,
 )
-from wagtail_vector_index.storage.base import VectorIndex
+from wagtail_vector_index.storage.base import Document, VectorIndex
 from wagtail_vector_index.storage.models import EmbeddingField
 
 fake = Faker()
 
 
+@pytest.fixture
+def mock_embedding_backend():
+    class MockEmbeddingBackend(BaseEmbeddingBackend):
+        def embed(self, texts):
+            def embedding_generator():
+                for text in texts:
+                    if "test" in text.lower():
+                        yield [1.0, 0.0, 0.0]
+                    elif "Very similar" in text:
+                        yield [0.9, 0.1, 0.0]
+                    elif "Somewhat similar" in text:
+                        yield [0.7, 0.3, 0.0]
+                    else:
+                        yield [0.1, 0.1, 0.8]
+
+            return embedding_generator()
+
+    return MockEmbeddingBackend
+
+
+@pytest.fixture
+def test_pages():
+    return [
+        ExamplePageFactory(title="Very similar to test"),
+        ExamplePageFactory(title="Somewhat similar to test"),
+        ExamplePageFactory(title="Not similar at all"),
+    ]
+
+
+@pytest.fixture
+def document_generator(test_pages):
+    def gen_documents(cls, *args, **kwargs):
+        for page in test_pages:
+            if "Very similar" in page.title:
```

Comment:
**Suggestion** Is it worth moving this to use the same mocked `embed` function so we don't have to repeat the defined vectors?

Reply:
Good suggestion. Done